imaging/tools/main.ts

```diff
@@ -167,13 +167,29 @@ const addTool = function (
   }
 };
 
+async function waitForImageId(element: EnabledElement) {
+  return new Promise((resolve, reject) => {
+    const startTime = Date.now(); // Record the start time
+    const check = () => {
+      if (element.image && element.image.imageId) {
+        resolve(element.image.imageId);
+      } else if (Date.now() - startTime >= 2000) {
+        // Check if 2000 ms timeout is reached
+        reject(new Error("Timeout: Image ID not available within 2000 ms"));
+      } else {
+        setTimeout(check, 50); // Check again in 100 milliseconds
+      }
+    };
+    check();
```

Comment:
Questa è da spiegare, non è mai bello usare i setTimeout :D

Reply:
lo so 😿  ho cercato di evitare ma purtroppo cornerstone.getEnabledElements() non ritorna una promise. 
A questo punto sarebbe meglio riscrivere la funzione in modo asincrono?